lib/tests/test.js

```diff
@@ -99,7 +99,44 @@ class Test {
     if (!this.sim) {
       this.sim = getSimulator();
     }
-    this.web3.setProvider(this.sim.provider(this.simOptions));
+
+    let simProvider = this.sim.provider(this.simOptions);
+
+    // Here we patch the sendAsync method on the provider. The goal behind this is to force pure/constant/view calls to become
+    // transactions, so that we can pull in execution traces and account for those executions in code coverage.
+    //
+    // Instead of a simple call, here's what happens:
+    //
+    // 1) A transaction is sent with the same payload, and a pre-defined gas price;
+    // 2) We wait for the transaction to be mined by asking for the receipt;
+    // 3) Once we get the receipt back, we dispatch the real call and pass the original callback;
+    //
+    // This will still allow tests to get the return value from the call and run contracts unmodified.
+    simProvider.realSendAsync = simProvider.sendAsync;
```

Comment:
Doing a bind is simpler and makes it so we don't have to have `apply` twice.
eg: `simProvider.realSendAsync = simProvider.sendAsync.bind(simProvider);`